starlette/types.py

```diff
@@ -1,9 +1,18 @@
+import sys
 import typing
 
-Scope = typing.MutableMapping[str, typing.Any]
-Message = typing.MutableMapping[str, typing.Any]
+if sys.version_info >= (3, 10):  # pragma: no cover
+    from typing import TypeAlias
+else:  # pragma: no cover
+    from typing_extensions import TypeAlias
 
-Receive = typing.Callable[[], typing.Awaitable[Message]]
-Send = typing.Callable[[Message], typing.Awaitable[None]]
+Scope: TypeAlias = typing.MutableMapping[str, typing.Any]
+Message: TypeAlias = typing.MutableMapping[str, typing.Any]
 
-ASGIApp = typing.Callable[[Scope, Receive, Send], typing.Awaitable[None]]
+Receive: TypeAlias = typing.Callable[[], typing.Awaitable[Message]]
+Send: TypeAlias = typing.Callable[[Message], typing.Awaitable[None]]
+
+ASGIApp: TypeAlias = typing.Callable[[Scope, Receive, Send], typing.Awaitable[None]]
+
+MiddewareType: TypeAlias = ASGIApp
```

Comment:
Is `TypeAlias` needed?

Reply:
See https://peps.python.org/pep-0613/. Here it does not add value, as it would be implicitly determined as a `TypeAlias` any way, but as `typing_extensions` is used in this project I figured why not use it, that way we know explicitly it is a `TypeAlias`.